internal/states/remote/state.go

```diff
@@ -280,6 +280,18 @@ func (s *State) Unlock(id string) error {
 	return nil
 }
 
+func (s *State) IsLockingEnabled() bool {
+	if s.disableLocks {
+		return false
+	}
+
+	if c, ok := s.Client.(OptionalClientLocker); ok {
+		lockInfo := c.IsLockingEnabled()
+		return lockInfo
+	}
+	return false
+}
```

Comment:
If `Client` implements `ClientLocker`, locking should be enabled:

```suggestion
func (s *State) IsLockingEnabled() bool {
	if s.disableLocks {
		return false
	}

        switch c := s.Client.(type) {
        // Client supports optional locking.
        case OptionalClientLocker:
                return c.IsLockingEnabled()
        // Client supports locking by default.
        case ClientLocker:
                return true
        // Client doesn't support any locking.
        default:
                return false
        }
}
```